internal/global/trace_test.go

```diff
@@ -24,22 +24,61 @@ import (
 
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/internal/global"
-	"go.opentelemetry.io/otel/oteltest"
 	"go.opentelemetry.io/otel/trace"
 )
 
-func TestTraceWithSDK(t *testing.T) {
+type fnTracerProvider struct {
+	tracer func(string, ...trace.TracerOption) trace.Tracer
+}
+
+func (fn fnTracerProvider) Tracer(instrumentationName string, opts ...trace.TracerOption) trace.Tracer {
+	return fn.tracer(instrumentationName, opts...)
+}
+
+type fnTracer struct {
+	start func(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span)
+}
+
+func (fn fnTracer) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
+	return fn.start(ctx, spanName, opts...)
+}
+
+func TestTraceProviderDelegation(t *testing.T) {
 	global.ResetForTest()
 
+	// Map of tracers to expected span names.
+	expected := map[string][]string{
+		"pre":      {"span2"},
+		"post":     {"span3"},
+		"fromSpan": {"span4"},
+	}
+
 	ctx := context.Background()
 	gtp := otel.GetTracerProvider()
 	tracer1 := gtp.Tracer("pre")
 	// This is started before an SDK was registered and should be dropped.
 	_, span1 := tracer1.Start(ctx, "span1")
 
-	sr := new(oteltest.SpanRecorder)
-	tp := oteltest.NewTracerProvider(oteltest.WithSpanRecorder(sr))
-	otel.SetTracerProvider(tp)
+	otel.SetTracerProvider(fnTracerProvider{
+		tracer: func(name string, opts ...trace.TracerOption) trace.Tracer {
+			spans, ok := expected[name]
+			assert.Truef(t, ok, "invalid tracer: %s", name)
+			return fnTracer{
+				start: func(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
+					if ok {
```

Comment:
Do we still need to check this since we are asserting it is true above?

Reply:
Without this you will not only get an error for an unknown tracer, but also an unknown span. Given it is implied you won't know spans from an unknown tracer the second error message will only add verbosity to the failure so I this conditional was added.